controller/config/config.go

```diff
@@ -187,52 +183,39 @@ func (c *Config) loadApiSection(edgeConfigMap map[interface{}]interface{}) error
 	c.Api.ActivityUpdateInterval = 90 * time.Second
 
 	if value, found := edgeConfigMap["api"]; found {
-		submap := value.(map[interface{}]interface{})
+		apiSubMap := value.(map[interface{}]interface{})
 
-		if value, found := submap["listener"]; found {
-			c.Api.Listener = value.(string)
-		} else {
-			return errors.New("required configuration value [edge.api.listener] missing")
-		}
-
-		if value, found := submap["advertise"]; found {
-			c.Api.Advertise = value.(string)
+		if val, ok := apiSubMap["address"]; ok {
+			if c.Api.Address, ok = val.(string); !ok {
+				return errors.Errorf("invalid type %t for [edge.api.address], must be string", val)
+			}
 		} else {
-			return errors.New("required configuration value [edge.api.advertise] missing")
-		}
-
-		var intValue = 0
-		if value, found := submap["sessionTimeoutMinutes"]; found {
-			intValue = value.(int)
-		}
-
-		if intValue < sessionTimeoutMin {
-			intValue = sessionTimeoutDefault
-			pfxlog.Logger().Warn("[edge.api.sessionTimeout] defaulted to " + strconv.Itoa(intValue))
+			return errors.New("required value [edge.api.address] is required")
 		}
 
-		c.Api.SessionTimeoutSeconds = time.Duration(intValue) * time.Minute
-
-		var apiIdentitySubMap map[interface{}]interface{}
-		if value, found = submap["identity"]; found {
-			apiIdentitySubMap = value.(map[interface{}]interface{})
+		var durationValue = 0 * time.Second
+		if value, found := apiSubMap["sessionTimeout"]; found {
+			strValue := value.(string)
+			durationValue, err = time.ParseDuration(strValue)
+			if err != nil {
+				return errors.Errorf("error parsing [edge.api.sessionTimeout], invalid duration string %s, cannot parse as duration (e.g. 1m): %v", strValue, err)
+			}
 		}
 
```

Comment:
loadIApiIdentity -> loadApiIdentity?

Reply:
What `loadIApiIdentity()` did is handled by xweb now. It creates the HTTP server and reads the proper root identity or override identity.